index/Dockerfile

```diff
@@ -14,21 +20,23 @@ RUN apt-get update \
         unzip \
         # Build tools\
         build-essential \
-        python3-pip \
         python3-dev \
         # For Psycopg2
-        libpq-dev\
+        libpq-dev \
         # Yaml parsing speedup, I think
         libyaml-dev \
         lsb-release \
         # for shapely with --no-binary
         libgeos-dev \
-        postgresql-client-14 \
+        postgresql-client-16 \
     # Cleanup
     && apt-get autoclean \
     && apt-get autoremove \
     && rm -rf /var/lib/{apt,dpkg,cache,log}
 
+ENV VIRTUAL_ENV /virtualenv/python3.10
+ENV PATH /virtualenv/python3.10/bin:$PATH
+
 COPY requirements.txt constraints.txt version.txt /conf/
 
 RUN cat /conf/version.txt \
```

Comment:
Can't you just add ` --break-system-packages` after `install` in the next few lines and skip everything else?

Reply:
I'd prefer not to as this feature in Ubuntu 24 was introduced for a reason.  Other than a few extra lines of code, is there any other drawbacks from using `virtualenv`?

Reply:
Thanks @SpacemanPaul Agree that the only real tangible benefit will be consistency on how the python runtime is used between both container and non-container environments, and hopefully make code maintenance easier in the future.